crypto/rsa/rsa_pss.c

```diff
@@ -191,6 +201,8 @@ int RSA_padding_add_PKCS1_PSS_mgf1(RSA *rsa, unsigned char *EM,
         if (RAND_bytes(salt, sLen) <= 0)
             goto err;
     }
+    if (hLen < 1 || hLen - 1 > emLen)
```

Comment:
But you we not concerned about possible overflow here, I guess?
If that is the case all the algebra here wins nothing, and it would be
far more easy to test the condition directly on the result.